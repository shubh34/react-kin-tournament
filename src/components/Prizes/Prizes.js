
import React from 'react';
import PropTypes from 'prop-types';
import Table from '../../sharedComponent/Table';
import { prizeBoardHeaders } from '../../configs/config';

const Prizes = ({ prizes }) => (
	<Table>

		<Table.Body>
			<Table.Row>
				{prizeBoardHeaders.map(header => <Table.Header key={header}>{header}</Table.Header>)}
			</Table.Row>
			{prizes.map(prize => (
				<Table.Row key={prize.fromPosition}>
					<Table.Column>
						{prize.range}
					</Table.Column>
					<Table.Column>
						<strong>
							{prize.prize}
						</strong>
					</Table.Column>
				</Table.Row>
			))}
		</Table.Body>
	</Table>
);


Prizes.propTypes = {
	prizes: PropTypes.arrayOf(PropTypes.shape({
		prize: PropTypes.string.isRequired,
		range: PropTypes.string.isRequired,
	})),
};

export default Prizes;
