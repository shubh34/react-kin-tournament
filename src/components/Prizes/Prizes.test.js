import React from 'react';
import { shallow } from 'enzyme';

import Prizes from './Prizes';
import { defaultPrizes } from '../../../tests/prizes';

const setUp = (setUpProps = {}, container = shallow) => {
	const props = {
		prizes: defaultPrizes,
		...setUpProps,
	};
	const wrapper = container(<Prizes {...props} />);
	return { wrapper, props };
};

describe('Prizes', () => {
	it('should render prizes', () => {
		const { wrapper } = setUp();
		expect(wrapper).toMatchSnapshot();
	});
});
