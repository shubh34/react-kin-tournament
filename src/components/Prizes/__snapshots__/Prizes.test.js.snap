// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prizes should render prizes 1`] = `
<Table>
  <TableBody>
    <TableRow>
      <TableHeader
        key="Pos"
      >
        Pos
      </TableHeader>
      <TableHeader
        key="Prize"
      >
        Prize
      </TableHeader>
    </TableRow>
    <TableRow
      key="1"
    >
      <TableColumn>
        1
      </TableColumn>
      <TableColumn>
        <strong>
          10836
        </strong>
      </TableColumn>
    </TableRow>
    <TableRow
      key="2"
    >
      <TableColumn>
        2-3
      </TableColumn>
      <TableColumn>
        <strong>
          5483
        </strong>
      </TableColumn>
    </TableRow>
    <TableRow
      key="4"
    >
      <TableColumn>
        4-5
      </TableColumn>
      <TableColumn>
        <strong>
          1079
        </strong>
      </TableColumn>
    </TableRow>
  </TableBody>
</Table>
`;
